fdbserver/EncryptKeyProxy.actor.cpp

```diff
@@ -89,6 +93,7 @@ using EncryptBaseCipherDomainIdKeyIdCache = std::unordered_map<EncryptBaseCipher
 
 struct EncryptKeyProxyData : NonCopyable, ReferenceCounted<EncryptKeyProxyData> {
 public:
+	Arena arena;
```

Comment:
Not familiar with arena, but I assume it can only grow in space, not shrinking? If so, how about store domainName in `Standalone<StringRef>` to avoid having an unbounded arena?

Reply:
Yeah, I don't think `shrinking` usecase is applicable here, I did think about Standalone, but went with specific Arena's to better surface out any leaks due to specific arena usage being high than common Standalone one getting charged for all encryption messages.

Reply:
My understanding is, with the arena being held at EncryptKeyProxyData level, it will grow unbounded, even after some of the StringRef hold in the cache is released. Standalone could avoid the leak.

Reply:
This Arena is at `EncryptKeyProxyData` level, which has the same lifetime as the EKP server. Or did I misread it?

Reply:
My bad, I mixed it with the EKP Request arena; yes Arena is EKProxyData level and is added for following reasons, earlier EKP cache only store `long`, however, with this change, it needs to store `StringRef` (domainName) as well, hence, while inserting `cell` into cache, StringRef ownership is assigned as EKProxyData level instead of EKP Request API arena (which would be incorrect to use).

However, the arena still shouldn't be ever increasing as EKP cache value updates (refresh ciphers periodically) would replace the value with a newer StringRef, hence, reclaiming the old one. At the moment, cache size is not bounded, but as discussed in past, there are plans to:
1. Bound the size of cache - implement LRU or LFU mechanism to reclaim space. 
2. Implement per cell timeout governed by KMS (this should control cache cell recycling frequency, however, number of cached elements still are governed by number of tenants supported by FDB cluster hosts). Also, I still believe updating to `Standalone<>` shouldn't change anything and having a EKP level Arena is better from debugging purposes. 
Thanks

Reply:
> ... replace the value with a newer StringRef, hence, reclaiming the old one.

I don't see how the old one can be reclaimed, as we are just using the same arena to allocate new memory for new cipher keys. Deleting object from from arena is a no-op.

But since the cache is already unbounded, and we know it will only grow slowly, it may be fine to have the underlying arena being unbounded. But once we move to have a bounded cache, it will need to change to Standalone or maybe FastAllocated for better memory management anyway.

Reply:
Agree, `Arena` here would be ever-growing. Though the growth of cache is slow and we plan to bound the cache-size, it feels `Standalone` is a better choice than EKProxyData context defined `Arena`. 

Will make necessary changes, thanks.